packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/event_client.dart

```diff
@@ -172,35 +174,38 @@ class EventClient implements Closeable {
           'putEvents - no EventsItemResponse response received from Pinpoint',
         );
       }
-      eventsItemResponse?.forEach((eventID, eventItemResponse) {
+      eventsItemResponse?.forEach((eventId, eventItemResponse) {
         if (!_equalsIgnoreCase(eventItemResponse.message ?? '', 'Accepted')) {
           _logger.warn(
-            'putEvents - issue with eventId: $eventID \n $eventItemResponse',
+            'putEvents - issue with eventId: $eventId \n $eventItemResponse',
           );
 
-          if (_isRetryable(eventItemResponse.message ?? '')) {
+          if (_isRetryable(eventItemResponse.statusCode)) {
             _logger.warn(
-              'putEvents - recoverable issue, will attempt to resend: $eventID in next FlushEvents',
+              'putEvents - recoverable issue, will attempt to resend: $eventId in next FlushEvents',
             );
```

Comment:
nit: reading this log message, it should be inside `if (_shouldEventRetry(eventId))`

Reply:
Good call thanks